UI/importers/classic.cpp

```diff
@@ -0,0 +1,577 @@
+		} else if (italic) {
+			font["style"] = "Italic";
+		} else {
+			font["style"] = "Regular";
+		}
+
+		settings["font"] = font;
+	} else if (id == "MonitorCaptureSource") {
+		out["id"] = "monitor_capture";
+
+		translate_int("monitor", in_settings, "monitor", settings, 0);
+		translate_bool("captureMouse", in_settings, "capture_cursor",
+			       settings);
+	} else if (id == "BitmapImageSource") {
+		out["id"] = "image_source";
+
+		string str = in_settings["path"].string_value();
+		settings["file"] = StringReplace(str, "\\\\", "/");
+	} else if (id == "BitmapTransitionSource") {
+		out["id"] = "slideshow";
+
+		Json files = in_settings["bitmap"];
+
+		if (!files.is_array()) {
+			files = Json::array{in_settings["bitmap"]};
+		}
+
+		settings["files"] = files;
+	} else if (id == "WindowCaptureSource") {
+		out["id"] = "window_capture";
+
+		string win = in_settings["window"].string_value();
+		string winClass = in_settings["windowClass"].string_value();
+
+		win = StringReplace(win, "/", "\\\\");
+		win = StringReplace(win, ":", "#3A");
+		winClass = StringReplace(winClass, ":", "#3A");
+
+		settings["window"] = win + ":" + winClass + ":";
+		settings["priority"] = 0;
+	} else if (id == "CLRBrowserSource") {
+		out["id"] = "browser_source";
+
+		string browser_dec =
+			QByteArray::fromBase64(in_settings["sourceSettings"]
+						       .string_value()
+						       .c_str())
+				.toStdString();
+
+		string err;
+
+		Json browser = Json::parse(browser_dec, err);
+
+		if (err != "")
+			return Json::object{};
+
+		Json::object obj = browser.object_items();
+
+		translate_string("CSS", obj, "css", settings);
+		translate_int("Height", obj, "height", settings, 0);
+		translate_int("Width", obj, "width", settings, 0);
+		translate_string("Url", obj, "url", settings);
+	} else if (id == "DeviceCapture") {
+		out["id"] = "dshow_input";
+
+		string device_id = in_settings["deviceID"].string_value();
+		string device_name = in_settings["deviceName"].string_value();
+
+		settings["video_device_id"] = device_name + ":" + device_id;
+
+		int w = in_settings["resolutionWidth"].int_value();
+		int h = in_settings["resolutionHeight"].int_value();
+
+		settings["resolution"] = to_string(w) + "x" + to_string(h);
+	} else if (id == "GraphicsCapture") {
+		bool hotkey = in_settings["useHotkey"].int_value() == 1;
+
+		if (hotkey) {
+			settings["capture_mode"] = "hotkey";
+		} else {
+			settings["capture_mode"] = "window";
+		}
+
+		string winClass = in_settings["windowClass"].string_value();
+		string exec = in_settings["executable"].string_value();
+
+		string window = ":" + winClass + ":" + exec;
+
+		settings["window"] = ":" + winClass + ":" + exec;
+
+		translate_bool("captureMouse", in_settings, "capture_cursor",
+			       settings);
+	}
+
+	out["settings"] = settings;
+
+	return out;
+}
+
+#undef translate_int
+#undef translate_string
+#undef translate_double
+#undef translate_bool
+
+static void translate_sc(Json::object in, Json *out)
```

Comment:
Should be something like:
```cpp
static void translate_sc(const Json::object &in, Json &out)
```
For the first parameter, if you don't use a reference, those objects will be copied. If you're not modifying the data, then they should also be const.  For the second parameter, you should not use pointers directly in C++ unless they're absolutely necessary, use a reference instead.